src/platforms/zephyr/src/libAtomVM/CMakeLists.txt

```diff
@@ -0,0 +1,175 @@
+#
+# This file is part of AtomVM.
+#
+# Copyright 2023 Winford (Uncle Grumpy) <winford@object.stream>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+#
+
```

Comment:
Honestly I don't like copying and pasting code. I opened a discussion about this topic: https://github.com/zephyrproject-rtos/zephyr/discussions/65704

Reply:
I completely agree that duplication of the CMakeLists.txt is not an ideal situation. Hopefully a solution will arise from the issue you opened, and I will continue to study the Zephyr documentation and search for a better solution.

Reply:
I also borrowed heavily from @adolfogc's comments in issue https://github.com/atomvm/AtomVM/issues/802#issuecomment-1711941648. I think we should give Adolfo E. García credit as well. I should probably reach out in a follow up to his post and ask for his explicit permission and see if he would like his name and contact info added to the copyright notice.

Reply:
I’m wondering if we should be setting standard c11 globally in cmake for each port, so that setting it again in `src/libAtomVM/CMakeLists.txt` would not be necessary. That way we could avoid any conflicts with toolchain specific methods for setting C11 as the standard for the library.

Reply:
@bettio @UncleGrumpy I think the some of this could be moved into the original `CMakeLists.txt` residing under `libAtomVM`, coupled with the use of boolean flags set accordingly for the platform (for example, `AVM_PLATFORM_ZEPHYR`). The project could be structured like a Zephyr module, similarly to how QCPP does.

Reply:
Assuming AtomVM is build as part of a Zephyr application, the variable could be set automatically like so:
```cmake
if(Zephyr_FOUND)
    set(AVM_PORT_ZEPHYR ON)
endif()
```

Reply:
Some of the checks could be avoided also. See how some of the auxiliary CMake macros are defined: https://github.com/zephyrproject-rtos/zephyr/blob/zephyr-v3.5.0/cmake/modules/extensions.cmake